### PR TITLE
Ws 2.11 add authorization

### PR DESCRIPTION
**Pull request by natashabuckham**
added config to Program.cs to AddDefaultTokenProviders() to Identity builder
added Authorize attribute to protected endpoints
tested on Postman with the header
added comment to backendClient.ts in the frontend folder for how to include the header on future fetch requests (the fetch requests there now are not meant to be authorized) [N.B. the login context is mentioned in the comments but is not present in this branch, still waiting to be merged into main]

Postman:
Screenshot 2024-09-10 103525
Screenshot 2024-09-10 103320
Screenshot 2024-09-10 103219

### CODE REVIEW

File: frontend/src/api/backendClient.ts [hunk 1]
```diff
@@ -23,3 +23,11 @@ export const registerUser = async (username: string, password: string) => {
...
   })
 }
+
+// to add the JWT token as a header to fetch requests which access protected endpoints do the following:
+// In the .tsx file where the fetch request is being called:
+// 1) import the login context to access the value of the JWT token
+// 2) pass the token as a prop called 'header' in the method which calls the fetch request
+// In this file:
+// 1) add 'header: string' as a prop to the method
+// 2) add "Authorization": `Bearer ${header}` to the list of headers
```
Comment (Kiran-R-K) [inline frontend/src/api/backendClient.ts:33]: Thanks for adding this, it's a nice clear explanation

**Review by Kiran-R-K — APPROVED** [review]
Really nice work - I'm approving this but can you hold off merging it for now, we'll want to do it in a way that that causes the least disruption as merging the ticket in is going to break all current api requests from frontend -> backend

**Comment by natashabuckham** [comment]
> Really nice work - I'm approving this but can you hold off merging it for now, we'll want to do it in a way that that causes the least disruption as merging the ticket in is going to break all current api requests from frontend -> backend

We could merge main into this branch and add the header to all the existing fetch requests in the frontend, and then merge in?

**Comment by Kiran-R-K** [comment]
> > Really nice work - I'm approving this but can you hold off merging it for now, we'll want to do it in a way that that causes the least disruption as merging the ticket in is going to break all current api requests from frontend -> backend
> 
> We could merge main into this branch and add the header to all the existing fetch requests in the frontend, and then merge in? @Kiran-R-K

yes please, that sounds great, go for it 👍 @natashabuckham @Anum1608

**Pull request merged**